### ISSUE TITLE
Decision Models: train your own Jev-style model in Bandits (tracking)

### ISSUE THREAD

**Issue by Alex-Hunterz** [bandr-ai/bandits#73]
## Goal

Anyone can train their own small Jev-style decision model inside Bandits: bring labeled decisions → score with an untrained model → SFT (LoRA) → calibrate → compare untrained vs trained vs Jev → try it in a local UI and API. We run it first on a public demo dataset, then launch.

A Jev-style model: an LLM reads a state, a question and 2–26 options, does one forward pass, and returns a probability for every option (softmax over option-letter logits). No generation.

## Locked decisions

- Everything in Bandits (`bandits/decide/`), heavy deps in an optional `decide` extra.
- **SFT only** (cross-entropy over option letters, LoRA). No RL.
- Recipe: Nimble-style core + AutoJev's discipline (option shuffling, best checkpoint on dev, temperature on its own split).
- First use case: bring your own labeled data. Judge distillation from traces is Track B, later.
- UI: one polished page (FastAPI + plain HTML/JS). No React, no Gradio.
- Splits: train / dev / calibration / locked test. Test is scored once, for the final report.
- All work on `feat/jev`; nothing merges to `main` without an explicit decision.

## Features (in order)

- [ ] #65 Generalize the decision dataset + import user data
- [ ] #66 Versioned prompt builder + untrained scorer
- [ ] #67 Pick the demo dataset + success bar
- [ ] #68 LoRA SFT trainer + go/no-go on dev
- [ ] #69 Temperature calibration + evaluation report
- [ ] #70 Model bundle + local decision API
- [ ] #71 One-page local UI
- [ ] #72 Cookbook, README, launch material

```text
#65 → #66 → #67 → #68 → #69 → #70 → #71 → #72
```

Already done: PR #64 (decision dataset compiler for judge verdicts + dense judge vote shares), unmerged.

## Open

- Base model: Qwen3.5-4B vs 9B (depends on GPU).
- Which GPU is available locally.
- Demo dataset: decided in #67.

## Later (Track B, not yet filed)

Train from Bandits agent traces: dataset inventory, human gold turn labels (TRAIL mapping + hand labels), targeted multi-vote judge runs + soft-label SFT, shared evidence view, baselines vs checks/judge, optional cascade judge backend.

**Comment by Alex-Hunterz** [comment]
**Branching and merge flow for Decision Models**

- `feat/jev` is the integration branch. PR #64 (`feat/jev` → `main`) stays open as the single umbrella PR.
- Work on this issue on its own branch off `feat/jev` (e.g. `feat/jev-<issue>-<name>`), with a PR **into `feat/jev`**, never `main`. Put `Closes #<issue>` in the PR body.
- If this issue depends on another issue's unmerged branch, branch from that one and retarget to `feat/jev` once it merges.
- Full test suite + ruff pass before opening the PR; each PR is reviewed before the next issue starts.
- Only a human merges. **PR #64 is merged into `main` once, after all Decision Models issues (#65–#72) are done.**
- Plan: [`docs/decision-models-plan.md`](https://github.com/bandr-ai/bandits/blob/feat/jev/docs/decision-models-plan.md).

**Comment by laxmanclo** [comment]
## Change of direction (2026-09-24): verifier → Jev leads the launch

**New headline:** your agent's traces → Bandits' verifiers label every step → train a small Jev on those labels → a fast, cheap copy of your verifier that can check every live step. It comes with an honest scorecard against the verifier, the untrained model and real Jev. "Bring your own labeled file" stays as the second entry point (CaseHOLD, #67).

This replaces "judge distillation is Track B, later". What we decided:

- **We trust the verifiers for now and improve them as we go.** The launch claims what's true today: *matches your verifier at a fraction of the cost*, not *catches every mistake*. When verifiers improve, relabel and retrain; the Jev improves with them.
- **Recipe, not core.** Everything lives in `recipes/jev/` (package `bandits_jev`, command `jev`). It reads core; core never imports it (#90).
- Speed over polish: the Jev wave is about 10 days old.

### Order

```text
#89 (trainer) ─► #90 recipe move ─► #91 split by trace ─► #92 scorecard ─► #94 one command ─► #95 GPU run ─► launch numbers
                                                              └► #93 cost/latency ┘
#98 judge more traces (API spend, parallel)      #97 Jev API client (needs key, parallel)
#96 Modal training ─► #71 UI ─► #70 bundle/API ─► #72 launch
```

| Issue | What | Suggested owner |
| --- | --- | --- |
| #89 | LoRA trainer: review and merge (stacked work depends on it) | Alex |
| #90 | Move into `recipes/jev` (done locally on `feat/jev-recipe`) | Laxman |
| #91, #92, #93, #94 | Split by trace, scorecard, cost/latency, one command + end-to-end test | Laxman |
| #95 | GPU script + first real run | Alex (rent the box) |
| #96 | Modal hosted training | Alex |
| #97 | Jev API client | whoever has the key |
| #98 | Judge more traces | either (Fireworks spend) |
| #70, #71, #72 | Bundle/API, UI, launch | after #95 |

Status of the original issues: #65/#66 merged via #74 (closing them was manual, since the PR went into `feat/jev`). #68 is #89. #69 is built and moved into the recipe on `feat/jev-recipe`. #67's converter and decision doc are there too; CaseHOLD is now the "bring your own file" demo, not the headline.
